src/service/ReservationService.java

```diff
@@ -0,0 +1,101 @@
+package service;
+
+import model.Customer;
+import model.IRoom;
+import model.Reservation;
+
+import java.util.*;
+
+public class ReservationService {
+    public static final ReservationService reservationService = new ReservationService();
+    private static final List<Reservation> reservations = new ArrayList<>();
+    private static final Map<String, IRoom> rooms = new HashMap<>();
+
+    public Collection<IRoom> findRooms(Date checkInDate, Date checkOutDate) {
+        List<IRoom> availableRooms = new ArrayList<>();
+        for (IRoom room : rooms.values()) {
+            if (reservationService.checkAvailability(room, checkInDate, checkOutDate)) {
+                availableRooms.add(room);
+            }
+        }
+        return availableRooms;
+    }
+
+    public Collection<IRoom> recommendedRooms(Date checkInDate, Date checkOutDate) {
+        System.out.println("No room found in the given date!");
+        Date checkInRecommended = addDaysToDate(checkInDate, 7);
+        Date checkOutRecommended = addDaysToDate(checkOutDate, 7);
+        System.out.println("Recommended room for 7 days later: ");
+        return findRooms(checkInRecommended, checkOutRecommended);
```

Comment:
This is very smartly programmed (reusing findRooms again here, but for a week later).